### ISSUE TITLE
Analyse Ekubo ETH/wstETH pool and suggest optimum LP range strategy

### ISSUE THREAD

**Issue by akiraonstarknet** [strkfarm/strategy-analysis#1]
## Context
1. We plan to create a strkfarm managed ETH/wstETH strategy on Ekubo. We need data analysis of past data to suggest LP parameters to manage investment with max profit
2. End usage: We will be providing certain liquidity to this pool in the suggested range and we will also rebalance the position from time to time to maintain maximum fee output while keeping impermanent loss less

## Methodology
1. Divide total available liquidity and swaps data into two parts: 70-30. 70% being used for training and optimisation, 30% used for testing the result

## Expected output
1. Analyse the liquidity and swaps data of ETH/wstETH pool with best TVL
2. Suggest logic for configuring lower & upper ticks for adding liquidity along with conditions to trigger rebalance
3. Show strategy performance on the test data

_

> ⚠️ **Note:** **Overfitting must be avoided. Do whatever optimisations u want on training data, once u run test on test data, share the result as it is. Do not optimise on test data.**

_

**Comment by ManvithaMolakala** [comment]
I would like to work on this issue.

**Comment by akiraonstarknet** [comment]
@ManvithaMolakala The project structure can be better. Broadly, u may need to modularize the project as
```yaml
- src
  - utils
    - ekubo_math.py # put general ekubo related formulae here (e.g. conversions, liquidity math, etc)
  - data
    - ekubo_data.py # write a class with various functions in a generic way that allows one to download data of liquidity, swaps for any pool
  - strategies
    - ekubo
      - base.py # used to write generic logic of strategy related code. Given any pool information, it can compute necessary things
      - eth_wsteth.py # uses functions from base.py, utils, etc. to present this pool specific strategy output
```

**Comment by Michaelkingsdev** [comment]
Hello @LazarusAA 
I am a frontend developer with 4 years of experience. I would really love to work on this task. Can it be assigned to me?

**Comment by Judah04** [comment]
Hi @akiraonstarknet I'd like to work on this task.
 I have experience using Python for Data analysis.
